Source/Core/Core/IOS/USB/Host.cpp

```diff
@@ -115,6 +116,18 @@ bool USBHost::UpdateDevices(const bool always_add_hooks)
 bool USBHost::AddNewDevices(std::set<u64>& new_devices, DeviceChangeHooks& hooks,
                             const bool always_add_hooks)
 {
+  if (Config::EmulateSkylanderPortal())
+  {
+    auto skylanderportal = std::make_unique<USB::SkylanderUSB>(m_ios, "Skylander Portal");
+    if (!ShouldAddDevice(*skylanderportal))
+      return true;
```

Comment:
This *probably* shouldn't return here, because that skips the regular USB devices below.

I'm also not entirely sure of the implications of mixing emulated and real USB devices here, but if it works it's probably fine...?

Reply:
Will write the opposite case, so for ShouldAddDevice rather than !ShouldAddDevice. I think it needed to be added in here so the USBHost can still track the device hooks, without the hooks it isn't aware there is a portal to send transfers to

Reply:
Just to circle back on this - when Dolphin isn't running deterministically this thread runs constantly to ensure that all new devices are added, undetected devices are removed and hooks are dispatched. Without adding the Emulated devices during this cycle, when added separately they are then detected as being removed and then not seen as an endpoint for Dolphin to communicate with. What I've done in my latest commit is extract the code in the AddDevices method and put it into a separate method, just so it's a bit tidier. Let me know if you have any other ideas for this :)